File: dp/05-Singleton/Singleton.java
public class Singleton
{
	private static Singleton _instance = null;

	private Singleton() {
		System.out.println("Creating instance...");
	}

	public static Singleton getInstance() {
		if (_instance == null) {
			_instance = new Singleton();
		}
		return _instance;
	}

	public static void doSomething() {
		System.out.println("Doing something...");
	}

	public static void main(String args[]) {
		Singleton.getInstance().doSomething();
		Singleton.getInstance().doSomething();
		Singleton.getInstance().doSomething();
	}

}

File: dp/05-Singleton/Singleton.cpp
#include "Singleton.h"
#include <iostream>

Singleton *Singleton::_instance = 0;

Singleton *Singleton::getInstance()
{
	if (_instance == 0) {
		_instance = new Singleton();
		std::cout << "creating singleton...\n";
	}
	return _instance;
}

void Singleton::doSomething()
{
	std::cout << "do something...\n";
}

int main()
{
	Singleton::getInstance()->doSomething();
	Singleton::getInstance()->doSomething();
	Singleton::getInstance()->doSomething();

	return 0;
}

